### DISCUSSION TITLE
Dati DDT in lista fatture ricevute (passive)

### DISCUSSION THREAD

**Discussion by pcapirci** [fattureincloud/api#463] [Q&A] [answered]
Salve, non mi è chiaro come fare a recuperare i dati relativi al/ai DDT di una fattura ricevuta (passiva); ovviamente se presenti. Mi riferisco nello specifico alla sezione 2.1.8 <DatiDDT> della specifica di tracciato; con i dati contenuti, relativi a NumeroDDT, DataDDT e RiferimentoNumeroLinea. Numero riferimento linea che, immagino, debba essere il campo id della collezione items_list (prego confermare).
Grazie mille
Paolo

**Comment by emajo** [comment]
Ciao @pcapirci, purtroppo al momento non mi è possibile replicare la sua casistica, il mio consiglio è quello di fare una chiamata [Get Received Document](https://developers.fattureincloud.it/api-reference#get-/c/-company_id-/received_documents/-document_id-) del documento in questione (con fieldset=detailed), per poi comparare la risposta con l'xml della fattura per trovare la corrispondenza dei campi che sta cercando.

Grazie,
Emanuele.

**Comment by pcapirci** [comment]
Buongiorno Emanuele e la ringrazio per la risposta,

ma io credo che sia compito del fornitore del servizio (in questo caso Fatture in Cloud) fornire la dovuta documentazione; perché appunto la mia domanda deve avere risposta tramite documentazione, e non tramite test come mi sta suggerendo.

Quello che richiedo, quindi, è un documento (ovviamente online) di mapping tra i campi della fattura XML, come descritta dall’Agenzia Entrate e quindi conforme, e i nomi campi delle vostre entità per come descritte nella vostra documentazione.

Non credo che tale informazione, ripeto di dominio del fornitore, debba essere lasciata ad un test del fruitore del servizio; che ovviamente, può anche sbagliarsi.

La ringrazio dell’approfondimento che richiedo.

Porgo cordiali saluti

Paolo Capirci

 

Da: Emanuele Saccomandi ***@***.***> 
Inviato: giovedì 25 luglio 2024 10:35
A: fattureincloud/api ***@***.***>
Cc: pcapirci ***@***.***>; Mention ***@***.***>
Oggetto: Re: [fattureincloud/api] Dati DDT in lista fatture ricevute (passive) (Discussion #463)

 

Ciao @pcapirci <https://github.com/pcapirci> , purtroppo al momento non mi è possibile replicare la sua casistica, il mio consiglio è quello di fare una chiamata Get Received Document <https://developers.fattureincloud.it/api-reference#get-/c/-company_id-/received_documents/-document_id->  del documento in questione (con fieldset=detailed), per poi comparare la risposta con l'xml della fattura per trovare la corrispondenza dei campi che sta cercando.

Grazie,
Emanuele.

—
Reply to this email directly, view it on GitHub <https://github.com/fattureincloud/api/discussions/463#discussioncomment-10146455> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AYPZS7X4LQ5HGEXOJBVNG2DZOC2BTAVCNFSM6AAAAABLMVNGMSVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMJUGY2DKNI> .
You are receiving this because you were mentioned.  <https://github.com/notifications/beacon/AYPZS7TBW5O4EADCM5IBUALZOC2BTA5CNFSM6AAAAABLMVNGMSWGG33NNVSW45C7OR4XAZNRIRUXGY3VONZWS33OINXW23LFNZ2KUY3PNVWWK3TUL5UWJTQATLJJO.gif> Message ID: ***@***.*** ***@***.***> >



-- 
Questa email è stata esaminata alla ricerca di virus dal software antivirus AVG.
www.avg.com

**Reply by valmoz** [reply, marked as answer]
Buongiorno @pcapirci 
non tutti i campi indicati nell'XML che riceviamo vengono effettivamente letti e riportati nelle nostre entità di Fatture in Cloud, questo è il caso anche dei campi che lei ha indicato.

Ha però la possibilità di **accedere all'XML originale del documento**, che le permetterà quindi di recuperare i valori richiesti. Per farlo dovrà effettuare una chiamata [Get Received Document](https://developers.fattureincloud.it/api-reference/#get-/c/-company_id-/received_documents/-document_id-) (oppure una [List Received Documents](https://developers.fattureincloud.it/api-reference/#get-/c/-company_id-/received_documents)) avendo l'accortezza di impostare il [Fieldset](https://developers.fattureincloud.it/docs/basics/additional-capabilities/customize-response) a "detailed"; in questo modo la risposta conterrà anche il campo **attachment_url**, che contiene l'URL da cui scaricare il file XML. A questo punto le basterà effettuare il parsing del contenuto per estrarre i valori che le servono.

Nota: l'attachment URL ha una **validità di 2 ore**, dopodiché sarà necessario ottenerne uno nuovo ripetendo la chiamata alle nostre API. Per questo motivo le consigliamo di salvare l'XML (o i dati al suo interno) e di non salvare direttamente l'URL che ritorniamo, in quanto risulterà inutilizzabile dopo la sua scadenza.

Grazie
Mauro

**Comment by pcapirci** [comment]
Grazie mille,

ritengo la risposta al momento esaustiva. Ovviamente dopo una fase di testing potrei aprire nuove richieste.

Questa la considero chiusa.

Ancora grazie

Paolo Capirci

 

Da: Mauro Valota ***@***.***> 
Inviato: venerdì 26 luglio 2024 14:35
A: fattureincloud/api ***@***.***>
Cc: pcapirci ***@***.***>; Mention ***@***.***>
Oggetto: Re: [fattureincloud/api] Dati DDT in lista fatture ricevute (passive) (Discussion #463)

 

Buongiorno @pcapirci <https://github.com/pcapirci> 
non tutti i campi indicati nell'XML che riceviamo vengono effettivamente letti e riportati nelle nostre entità di Fatture in Cloud, questo è il caso anche dei campi che lei ha indicato.

Ha però la possibilità di accedere all'XML originale del documento, che le permetterà quindi di recuperare i valori richiesti. Per farlo dovrà effettuare una chiamata Get Received Document <https://developers.fattureincloud.it/api-reference/#get-/c/-company_id-/received_documents/-document_id->  (oppure una List Received Documents <https://developers.fattureincloud.it/api-reference/#get-/c/-company_id-/received_documents> ) avendo l'accortezza di impostare il Fieldset <https://developers.fattureincloud.it/docs/basics/additional-capabilities/customize-response>  a "detailed"; in questo modo la risposta conterrà anche il campo attachment_url, che contiene l'URL da cui scaricare il file XML. A questo punto le basterà effettuare il parsing del contenuto per estrarre i valori che le servono.

Nota: l'attachment URL ha una validità di 2 ore, dopodiché sarà necessario ottenerne uno nuovo ripetendo la chiamata alle nostre API. Per questo motivo le consigliamo di salvare l'XML (o i dati al suo interno) e di non salvare direttamente l'URL che ritorniamo, in quanto risulterà inutilizzabile dopo la sua scadenza.

Grazie
Mauro

—
Reply to this email directly, view it on GitHub <https://github.com/fattureincloud/api/discussions/463#discussioncomment-10159750> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AYPZS7TVL6UKSQ3U267YTIDZOI7A3AVCNFSM6AAAAABLMVNGMSVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMJVHE3TKMA> .
You are receiving this because you were mentioned.  <https://github.com/notifications/beacon/AYPZS7RG27LSKYZ3WX5FXALZOI7A3A5CNFSM6AAAAABLMVNGMSWGG33NNVSW45C7OR4XAZNRIRUXGY3VONZWS33OINXW23LFNZ2KUY3PNVWWK3TUL5UWJTQATMDIM.gif> Message ID: ***@***.*** ***@***.***> >



-- 
Questa email è stata esaminata alla ricerca di virus dal software antivirus AVG.
www.avg.com
